nsv13/code/modules/overmap/physics.dm

```diff
@@ -78,8 +79,29 @@
 /obj/structure/overmap/proc/can_move()
 	return TRUE //Placeholder for everything but fighters. We can later extend this if / when we want to code in ship engines.
 
-/obj/structure/overmap/process(time)
-	time /= 1 SECONDS
+/*
+
+This proc allows overmaps to survive even in the laggiest of server conditions. Overmaps MUST always process, and sometimes the game will throttle the subsystems during lag. We cannot, however, live without them. That's where this beauty comes into play.
+Basically, when we process, we store the last time we were able to be processed. If the last time we were processed was 1 or more seconds ago, then we take control of our own processing with a while() loop.
+The while loop runs at a programatic level and is thus separated from any throttling that the server may put in place. 5 minutes after starting the failsafe processing, we'll see if the game is ready to take back control or not.
+
+*/
+
+/obj/structure/overmap/proc/start_failsafe_processing()
+	set waitfor = FALSE //Don't hang the process call.
+	processing_failsafe = TRUE
+	addtimer(VARSET_CALLBACK(src, processing_failsafe, FALSE), 10 MINUTES) //At this point, the game is under immense strain. In a few minutes time we'll attempt to hand back control to processing, but for now, we're going to handle it ourselves.
+	while(processing_failsafe)
+		stoplag() //Lock up the thread for a bit, throttle the process down to whatever the server can handle right now.
+		if(last_process < world.time - 0.5 SECONDS)
+			process()
+
+/obj/structure/overmap/process()
+	set waitfor = FALSE
+	var/time = min(world.time - last_process, 10)
+	time /= 10 // fuck off deciseconds
+	if(last_process > 0 && (last_process < world.time - 1 SECONDS) && !processing_failsafe) //Alright looks like the game's shat itself. Time to engage "failsafe mode". The logic of this is that if we've not been processed for over 1 second, then ship piloting starts to become unbearable and we need to step in and do our own processing, until the game's back on its feet again.
+		start_failsafe_processing()
```

Comment:
..... This should be in a specific OVERMAP PROCESSING SUBSYSTEM. NOT PER DAMN OBJECT!

Reply:
This misses the entire point of this change.